spirv_hlsl.cpp

```diff
@@ -2994,35 +2993,45 @@ void CompilerHLSL::emit_texture_op(const Instruction &i, bool sparse)
 
 	if (hlsl_options.shader_model < 40)
 	{
-		string coord_filler;
-		uint32_t modifier_count = 0;
-
-		if (lod)
+		if (dref)
 		{
+			if (imgtype.image.dim != spv::Dim1D && imgtype.image.dim != spv::Dim2D)
+				SPIRV_CROSS_THROW("Depth comparison is only supported for 1D and 2D textures in HLSL shader model 2/3.");
+
+			if (grad_x || grad_y)
+				SPIRV_CROSS_THROW("Depth comparison is not supported for grad sampling in HLSL shader model 2/3.");
+
+			forward = forward && should_forward(dref);
 			for (uint32_t size = coord_components; size < 3; ++size)
-				coord_filler += ", 0.0";
-			coord_expr = "float4(" + coord_expr + coord_filler + ", " + to_expression(lod) + ")";
-			modifier_count++;
+				coord_expr += ", " + to_expression(dref);
```

Comment:
Should dref be duplicated twice out to 3 components when using tex1D?

Reply:
I guess it doesn't hurt though, but ...

Reply:
Not necessary, just matching the behaviour of the NV Cg compiler, which put the depth in both y and z.  My guess was that it might have been one less instruction to just swizzle than having to add in a 0?  No idea.  Should I change it?

Reply:
I would normally expect the second coordinate to be 0 for 1D, but if Cg does it that way I don't mind it being there as-is. Just add a comment I guess to not confuse future readers.